sqlglot/expressions.py

```diff
@@ -6793,7 +6793,7 @@ def parse_identifier(name: str | Identifier, dialect: DialectType = None) -> Ide
     return expression
 
 
-INTERVAL_STRING_RE = re.compile(r"\s*([0-9]+)\s*([a-zA-Z]+)\s*")
+INTERVAL_STRING_RE = re.compile(r"\s*(\d+)\s*([a-zA-Z]+)\s*")
```

Comment:
Is `\d` matching new test cases better than `[0-9]` ?

Reply:
No, I used it to make the regex more compact but I actually reverted since it captures a superset of `[0-9]` (eg. weird unicode numerals) that we don't care about.